### ISSUE TITLE
F-Droid 5.2 crashes

### ISSUE THREAD

**Issue by licaon-kter** [simplex-chat/simplex-chat#2790]
Android 13

_Will get log later_

**Comment by shumvgolove** [comment]
LineageOS 20 (Android 13), aarch64

Seems working fine for me?
![Screenshot_20230728-091255_SimpleX](https://github.com/simplex-chat/simplex-chat/assets/37271604/93814682-0b7e-4ce6-9f9e-037974795577)

**Comment by licaon-kter** [comment]
log: https://gist.github.com/licaon-kter/baa5b4c37153790bc7c3166ebefc50c2

Is this https://github.com/simplex-chat/simplex-chat/issues/959 and needs to be reopened?

I've closed all the other apps (except keyboard and launcher) and still won't start. System was reporting 1.4Gb RAM free at least.

**Comment by shumvgolove** [comment]
Hmm, let me try to back up database and perform full reinstallation to double-check.

**Comment by shumvgolove** [comment]
Strange, installs and works fine for me... 

> Is this https://github.com/simplex-chat/simplex-chat/issues/959 and needs to be reopened?

Not sure. Your log looks similar, but at that time F-Droid build didn't work for me too. 

Do you use LineageOS Nightly builds? If there's any update for your device, could you perhaps perform an update?

**Comment by licaon-kter** [comment]
https://divestos.org/pages/news I'm already on July 20 build so no.

You figure it's an Android distro issue? Let me ~~blame~~ping @SkewedZeppelin then :)

**Comment by shumvgolove** [comment]
Oh, you're using DivestOS?

Then it's certainly an issue with memory allocation management, cause DivestOS use [`hardened_malloc`](https://github.com/GrapheneOS/hardened_malloc) from GrapheneOS. Relevant line from your log:
```
 21416 W Adreno-GSL: <sharedmem_gpuobj_alloc:2742>: sharedmem_gpumem_alloc: mmap failed errno 12 Out of memory
 21416 E Adreno-GSL: <gsl_memory_alloc_pure:2583>: GSL MEM ERROR: kgsl_sharedmem_alloc ioctl failed.
 21416 W libEGL  : eglInitialize(0x40227efc40) failed (EGL_BAD_ALLOC)
 21416 W OpenGLRenderer: Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
 21416 W OpenGLRenderer: Failed to initialize 101010-2 format, error = EGL_NOT_INITIALIZED
 21416 F OpenGLRenderer: Failed to create context, error = EGL_NOT_INITIALIZED
```

I was having the same issue with using proot distros in `Tmux` 🙃

**Comment by licaon-kter** [comment]
Hope I don't get between Android OS devs and Android app devs


ahem https://github.com/Divested-Mobile/DivestOS-Build/issues/220#issuecomment-1624134113 :)

**Comment by shumvgolove** [comment]
I will check 5.2 (from F-Droid) on my other device with GrapheneOS later this day. It appears that apk from GitHub releases works fine in GrapheneOS 🤔 Could you check if it works @licaon-kter ?

If it is the case, then there's something needs to be fixed with `build-android.sh`/`nix build`.

**Comment by licaon-kter** [comment]
`simplex.apk` from https://github.com/simplex-chat/simplex-chat/releases/tag/v5.2.0 crashes too

**Comment by shumvgolove** [comment]
Thanks for confirmation! Will investigate.

**Comment by licaon-kter** [comment]
Of note, another device, but same OS, 5.1.3 crashes also.

**Comment by SkewedZeppelin** [comment]
DivestOS lacks the per app toggle for hmalloc like GrapheneOS.
As a workaround you can install the 32-bit variant, which will use scudo instead of hmalloc.

**Comment by pjanx** [comment]
5.2.1 crashes on launch, F-Droid, Android 8.1, LineageOS 15.1

**Comment by licaon-kter** [comment]
@pjanx can you grab a logcat? Say via ADB or Scoop app?

**Comment by pjanx** [comment]
I just have, it seems to be #2689. Sorry for the noise.

**Comment by licaon-kter** [comment]
5.4.2 arm64 worked fine on Divest with "secure app spawn" enabled... **but only once** after install... then I disabled spawining, crash, then enabled spawning, crash, now it never works, even reinstalled. What the...

**Comment by rainbowforgedev** [comment]
This may be unrelated to this issue, but I'm on lineage 21, and I was able to use SimpleXchat for a couple weeks, but in late December 24 with an update the app started crashing in a similar way described in the above comments. Once I have a chance to get a logcat I may open a new issue with that. In the meantime, I'm going to try to roll back versions and find out which was the last one that actually worked.
